python/neutron-understack/neutron_understack/nautobot.py

```diff
@@ -1,77 +1,78 @@
-    message = "%(obj)s not found in Nautobot. ref=%(ref)s"
+class NautobotOSError(exc.NeutronException):
+    message = "Error occurred querying Nautobot: %(err)s"
 
 
-class Nautobot:
-    CALLER_FRAME = 1
+class NautobotNotFoundError(exc.NeutronException):
+    message = "%(obj)s not found in Nautobot. ref=%(ref)s"
 
-    def __init__(self, nb_url: str | None = None, nb_token: str | None = None):
-        """Basic Nautobot wrapper because pynautobot doesn't expose plugin APIs."""
-        self.base_url = nb_url or "http://nautobot-default.nautobot.svc.cluster.local"
-        self.token = nb_token or self._fetch_nb_token()
-        self.s = requests.Session()
-        self.s.headers.update({"Authorization": f"Token {self.token}"})
 
-    def _fetch_nb_token(self):
-        file = pathlib.Path("/etc/nb-token/token")
-        with file.open() as f:
-            return f.read().strip()
+class Nautobot:
+    """Basic Nautobot wrapper because pynautobot doesn't expose plugin APIs."""
 
-    def _log_and_raise_for_status(self, response):
-        try:
-            response.raise_for_status()
-        except HTTPError as error:
-            LOG.error("Nautobot error: %s %s", error, response.content)
-            raise NautobotError() from error
+    def __init__(self, nb_url: str, nb_token: str):
+        self.base_url = nb_url
+        self.session = requests.Session()
+        self.session.headers.update({"Authorization": f"Token {nb_token}"})
 
     def make_api_request(
-        self, url: str, method: str, payload: dict | None = None, params=None
+        self,
+        method: str,
+        url: str,
+        payload: dict | None = None,
+        params: dict[str, str] | None = None,
+        timeout: int = 10,
     ) -> dict:
-        endpoint_url = urljoin(self.base_url, url)
-        caller_function = inspect.stack()[self.CALLER_FRAME].function
-        http_method = method.upper()
+        full_url = urljoin(self.base_url, url)
 
-        LOG.debug(
-            "%(caller_function)s payload: %(payload)s",
-            {"payload": payload, "caller_function": caller_function},
-        )
-        resp = self.s.request(
-            http_method,
-            endpoint_url,
-            timeout=10,
-            json=payload,
-            params=params,
-            allow_redirects=False,
-        )
-
-        if resp.content:
-            resp_data = resp.json()
-        else:
-            resp_data = {"status_code": resp.status_code}
+        try:
+            response = self.session.request(
+                method,
+                full_url,
+                timeout=timeout,
+                json=payload,
+                params=params,
+                allow_redirects=False,
+            )
+        except Exception as e:
+            raise NautobotOSError(err=e) from e
+
+        if response.status_code >= 300:
+            raise NautobotRequestError(
+                code=response.status_code, url=full_url, body=response.content
```

Comment:
```suggestion
                code=response.status_code, url=full_url, body=response.content
```

long term you'll probably want the `response.json()["error"]` by default.